StardewArchipelago/GameModifications/CodeInjections/Modded/JunimoShopInjections.cs

```diff
@@ -0,0 +1,224 @@
+            catch (Exception ex)
+            {
+                _monitor.Log($"Failed in {nameof(Update_JunimoWoodsAPShop_Prefix)}:\n{ex}", LogLevel.Error);
+                return true; // run original logic
+            }
+        }
+
+                private class PurpleJunimo
+        {
+            public StardewItem OfferedItem {get; set;}
+            public int Amount {get; set;}
+        }
+
+        public static void PurpleJunimoSpecialShop()
+        {
+            var junimoWoods = Game1.player.currentLocation;
+            var friendsMet = Game1.player.friendshipData.Count();
+            var fakeStock = new Dictionary<string, int>(){
+                {"Money", 10000},
+                {"Dewdrop", 2500},
+                {"Friendship", 200 * friendsMet}
+            };
+            PurpleJunimoOptions = new Dictionary<string, PurpleJunimo>();
+            var randomSeed = 0;
+            foreach (var item in fakeStock)
+            {
+                var currentMonth = (int)(Game1.stats.daysPlayed / 28) + 1;
+                var random = new Random((int)Game1.uniqueIDForThisGame / 2 + currentMonth + randomSeed);
+                var itemToTrade = _junimoShopGenerator.PurpleItems.ElementAt(random.Next(_junimoShopGenerator.PurpleItems.Count));
+                var stardewItem = _stardewItemManager.GetObjectById(itemToTrade.Key);
+                var itemToTradeTotal = JunimoShopGenerator.ExchangeRate(item.Value, itemToTrade.Value);
+                PurpleJunimoOptions[item.Key] = new PurpleJunimo(){
+                    OfferedItem = stardewItem,
+                    Amount = Math.Max(itemToTradeTotal[1] / itemToTradeTotal[0], 1),
+                };
+                randomSeed += 1;
+            }
+            junimoWoods.createQuestionDialogue(
+                    Question,
+                    new Response[4]
+                    {
+                        new("Money", string.Format(Money, PurpleJunimoOptions["Money"].Amount.ToString(), PurpleJunimoOptions["Money"].OfferedItem.Name)),
+                        new("Dewdrop", string.Format(Dewdrop, PurpleJunimoOptions["Dewdrop"].Amount.ToString(), PurpleJunimoOptions["Dewdrop"].OfferedItem.Name)),
+                        new("Friendship", string.Format(Friendship, PurpleJunimoOptions["Friendship"].Amount.ToString(), PurpleJunimoOptions["Friendship"].OfferedItem.Name)),
+                                                Nothing
+
+                    }, _junimoDialogueKey);
+
+
+            return;
+
+        }
+        
+        public static bool AnswerDialogueAction_Junimoshop_Prefix(GameLocation __instance, string questionAndAnswer, string[] questionParams, ref bool __result)
+        {
+            try
+            {
+                if (!questionAndAnswer.StartsWith(_junimoDialogueKey))
+                {
+                    return true; // run original logic
+                }
+                var answer = questionAndAnswer.Split("_")[1];
+                if (answer == "No")
```

Comment:
Every if block in this method could have its content be a sub method

Reply:
I did some of them just not all.  If its 2~3 lines I didn't.